eng/common/scripts/Validate-Package.ps1

```diff
@@ -0,0 +1,312 @@
+#This script is responsible for release preparedness check that's run as part of build pipeline.
+
+[CmdletBinding()]
+param (
+  [Parameter(Mandatory = $true)]  
+  [string] $PackageName,
+  [Parameter(Mandatory = $true)] 
+  [string] $ArtifactPath,
+  [Parameter(Mandatory=$True)]
+  [string] $RepoRoot,
+  [Parameter(Mandatory=$True)]
+  [string] $APIKey,  
+  [Parameter(Mandatory=$True)]
+  [string] $ConfigFileDir,
+  [string] $BuildDefinition,
+  [string] $PipelineUrl,
+  [string] $APIViewUri,
+  [string] $Devops_pat = $env:DEVOPS_PAT
+)
+Set-StrictMode -Version 3
+
+. (Join-Path $PSScriptRoot common.ps1)
+. ${PSScriptRoot}\Helpers\ApiView-Helpers.ps1
+. ${PSScriptRoot}\Helpers\DevOps-WorkItem-Helpers.ps1
+
+if (!$Devops_pat) {
+  az account show *> $null
+  if (!$?) {
+    Write-Host 'Running az login...'
+    az login *> $null
+  }
+}
+else {
+  # Login using PAT
+  LoginToAzureDevops $Devops_pat
+}
+
+az extension show -n azure-devops *> $null
+if (!$?){
+  az extension add --name azure-devops
+} else {
+  # Force update the extension to the latest version if it was already installed
+  # this is needed to ensure we have the authentication issue fixed from earlier versions
+  az extension update -n azure-devops *> $null
+}
+
+CheckDevOpsAccess
+
+class ValidationStatus
+{
+    [string]$Name
+    [string]$Status
+    [string]$Message
+
+    ValidationStatus([string]$name, [string]$status)
+    {
+        $this.Name = $name
+        $this.Status = $status
+    }
+}
+
+class PackageDetails
+{
+    [string]$Name
+    [string]$Version
+    [ValidationStatus]$VersionValidation
+    [ValidationStatus]$ChangeLogValidation
+    [ValidationStatus]$APIReviewValidation
+    [ValidationStatus]$PackageNameValidation
+
+    PackageDetails([string]$name)
+    {
+        $this.Name = $name
+    }
+}
+
+function ValidateVersion($packageName, $versionString)
+{
+    $validationStatus = [ValidationStatus]::new("Version Validation", "Success")
+    $version = [AzureEngSemanticVersion]::ParseVersionString($versionString)
```

Comment:
Why are we validating the version? If we do validate it, shouldn't we do something like ensure it hasn't been released or something along those lines?

Reply:
Current validation is just to ensure version is in expected format. We don't create any status update in package work item if it finds that version was already shipped.

Reply:
Removed version validation.